src/Workspaces/Core/Portable/Execution/ChecksumTreeCollection.TreeNodes.cs

```diff
@@ -121,6 +156,39 @@ public virtual ChecksumObject TryGetChecksumObject(Checksum checksum, Cancellati
                 return null;
             }
 
+            public virtual void AppendChecksumObjects(ImmutableDictionary<Checksum, ChecksumObject>.Builder builder, HashSet<Checksum> checksums, CancellationToken cancellationToken)
```

Comment:
sure.

it is virtual since Root Node has extra checking for additional assets.